### PR TITLE
modulesync 0.12.8

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
@daenney do you have any thoughts about the failing tests?

**Comment by daenney** [comment]
@bastelfreak You broke a thing or two 😛

**Comment by bastelfreak** [comment]
Ha found it. The syntax check throws an obsucre error if I use an instance variable
```
$ bundle exec rake syntax:templates
/home/bastelfreak/modulesync_config/modules/puppet-facette/.vendor/ruby/2.3.0/gems/yard-0.9.5/lib/yard/code_objects/base.rb:51: warning: already initialized constant YARD::CodeObjects::CONSTANTMATCH
/home/bastelfreak/modulesync_config/modules/puppet-facette/.vendor/ruby/2.3.0/gems/puppet-strings-0.4.0/lib/puppet_x/puppetlabs/strings/yard/core_ext/yard.rb:10: warning: previous definition of CONSTANTMATCH was here
/home/bastelfreak/modulesync_config/modules/puppet-facette/.vendor/ruby/2.3.0/gems/yard-0.9.5/lib/yard/code_objects/base.rb:54: warning: already initialized constant YARD::CodeObjects::CONSTANTSTART
/home/bastelfreak/modulesync_config/modules/puppet-facette/.vendor/ruby/2.3.0/gems/puppet-strings-0.4.0/lib/puppet_x/puppetlabs/strings/yard/core_ext/yard.rb:21: warning: previous definition of CONSTANTSTART was here
---> syntax:templates
rake aborted!
JSON::GeneratorError: only generation of JSON objects or arrays allowed
/home/bastelfreak/modulesync_config/modules/puppet-facette/.vendor/ruby/2.3.0/gems/json_pure-1.8.3/lib/json/common.rb:285:in `generate'
/home/bastelfreak/modulesync_config/modules/puppet-facette/.vendor/ruby/2.3.0/gems/json_pure-1.8.3/lib/json/common.rb:285:in `pretty_generate'
templates/facette.json.erb:1:in `<main>'
/home/bastelfreak/modulesync_config/modules/puppet-facette/.vendor/ruby/2.3.0/gems/puppet-syntax-2.1.0/lib/puppet-syntax/templates.rb:52:in `validate_erb'
/home/bastelfreak/modulesync_config/modules/puppet-facette/.vendor/ruby/2.3.0/gems/puppet-syntax-2.1.0/lib/puppet-syntax/templates.rb:18:in `block in check'
/home/bastelfreak/modulesync_config/modules/puppet-facette/.vendor/ruby/2.3.0/gems/puppet-syntax-2.1.0/lib/puppet-syntax/templates.rb:14:in `check'
/home/bastelfreak/modulesync_config/modules/puppet-facette/.vendor/ruby/2.3.0/gems/puppet-syntax-2.1.0/lib/puppet-syntax/tasks/puppet-syntax.rb:58:in `block (2 levels) in initialize'
/home/bastelfreak/modulesync_config/modules/puppet-facette/.vendor/ruby/2.3.0/gems/rake-11.3.0/exe/rake:27:in `<top (required)>'
/home/bastelfreak/.gem/ruby/2.3.0/gems/bundler-1.12.5/lib/bundler/cli/exec.rb:63:in `load'
/home/bastelfreak/.gem/ruby/2.3.0/gems/bundler-1.12.5/lib/bundler/cli/exec.rb:63:in `kernel_load'
/home/bastelfreak/.gem/ruby/2.3.0/gems/bundler-1.12.5/lib/bundler/cli/exec.rb:24:in `run'
/home/bastelfreak/.gem/ruby/2.3.0/gems/bundler-1.12.5/lib/bundler/cli.rb:304:in `exec'
/home/bastelfreak/.gem/ruby/2.3.0/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/home/bastelfreak/.gem/ruby/2.3.0/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/home/bastelfreak/.gem/ruby/2.3.0/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
/home/bastelfreak/.gem/ruby/2.3.0/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
/home/bastelfreak/.gem/ruby/2.3.0/gems/bundler-1.12.5/lib/bundler/cli.rb:11:in `start'
/home/bastelfreak/.gem/ruby/2.3.0/gems/bundler-1.12.5/exe/bundle:27:in `block in <top (required)>'
/home/bastelfreak/.gem/ruby/2.3.0/gems/bundler-1.12.5/lib/bundler/friendly_errors.rb:98:in `with_friendly_errors'
/home/bastelfreak/.gem/ruby/2.3.0/gems/bundler-1.12.5/exe/bundle:19:in `<top (required)>'
/home/bastelfreak/.gem/ruby/2.3.0/bin/bundle:23:in `load'
/home/bastelfreak/.gem/ruby/2.3.0/bin/bundle:23:in `<main>'
Tasks: TOP => syntax:templates
(See full trace by running task with --trace)
```

**Comment by bastelfreak** [comment]
I am not really sure about the root cause here. I can run `bundle exec rake spec` just fine, something seems to be wrong with the template.

**Comment by daenney** [comment]
The template seems to work just fine on my machines though.

**Comment by daenney** [comment]
Alright so the reason the validation of the templates breaks is:

* We call `JSON.pretty_generate()` in the templates b/c surprise, we need to generate JSON
* `.pretty_generate()` only accepts object or array as argument, but during the validation phase the config variable is not set, aka `nil` and therefor it explodes

There's two ways we can fix it:
* Disable the validation of templates with `.json` in the extension. Since all they do is call `JSON.pretty_generate()` and the tests validate that this works the syntax check won't actually catch anything here
* Do some stuff in Puppet to generate the JSON there and then simply write the hash in the ERB instead.

File: spec/classes/facette_spec.rb [hunk 1]
```diff
@@ -1,178 +1,212 @@
...
-            'lsbmajdistrelease'   => '14.04',
-            'lsbdistdescription'  => 'Ubuntu 14.04.3 LTS',
-            'lsbdistid'           => 'Ubuntu',
-            :osfamily             => 'Debian',
-            :operatingsystem      => 'Ubuntu',
-        } }
+  describe 'on Ubuntu' do
+    let(:facts) do
+      {
+        'lsbdistcodename' => 'trusty',
```
Comment (bbriggs) [inline spec/classes/facette_spec.rb]: Align this hash rocket

**Pull request merged**